doc/api/deprecations.md

```diff
@@ -2214,6 +2214,22 @@ the `_handle` property of the `Cipher`, `Decipher`, `DiffieHellman`,
 Using the `_handle` property to access the native object is deprecated because
 improper use of the native object can lead to crashing the application.
 
+<a id="DEP0XXX"></a>
+### DEP0XXX: dns.lookup() supports for a falsy hostname
+<!-- YAML
+changes:
+  - version: REPLACEME
+    pr-url: https://github.com/nodejs/node/pull/23173
+    description: Runtime deprecation.
+-->
+
+Type: Runtime
+
+Previous versions of Node.js supported `dns.lookup()` with a falsy hostname
+like `dns.lookup(false)` due to backward compatibility long before.
+This behavior is undocumented and seems useless in real world apps. We
+might make passing invalid hostname throw in the future.
```

Comment:
I would replace "seems useless" with "is thought to be unused" and strengthen the second sentence to: "It will become an error in future versions of Node.js."

We might never do that but it's better to reserve the option and leave no room for doubt.